File: pr1-java/networking/src/main/java/pr1Java/networking/reflection/Response.java
package pr1Java.networking.reflection;

import java.io.Serializable;

public class Response implements Serializable {
    private ResponseType type;
    private Object data;

    private Response() {
    }

    public ResponseType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }

    private void setType(ResponseType type) {
        this.type = type;
    }

    private void setData(Object data) {
        this.data = data;
    }

    @Override
    public String toString() {
        return "Response{" +
                "type='" + type + '\'' +
                ", data='" + data + '\'' +
                '}';
    }

    public static class Builder {
        private final Response response = new Response();

        public Builder setType(ResponseType type) {
            response.setType(type);
            return this;
        }

        public Builder setData(Object data) {
            response.setData(data);
            return this;
        }

        public Response build() {
            return response;
        }
    }

}

File: pr1-cs/Networking/Reflection/Response.cs
using System;

namespace Networking.Reflection
{
    [Serializable]
    public class Response
    {
        public ResponseType Type { get; set; }
        public object Data { get; set; }

        public Response(ResponseType type)
        {
            Type = type;
            Data = null;
        }

        public Response(ResponseType type, object data)
        {
            Type = type;
            Data = data;
        }

        public Response SetType(ResponseType type)
        {
            Type = type;
            return this;
        }

        public Response SetData(object data)
        {
            Data = data;
            return this;
        }
    }
}